src/metadata-parser.h

```diff
@@ -222,6 +222,95 @@ namespace librealsense
         };
     };
 
+
+	class actual_fps_calculator
+	{
+	public:
+		rs2_metadata_type get_fps(const frame & frm)
+		{
+			auto num_of_frames = 0;
+			if (frm.additional_data.frame_number < frm.additional_data.last_frame_number)
+			{
+				num_of_frames = std::numeric_limits<unsigned long long>::max() - frm.additional_data.last_frame_number + frm.additional_data.frame_number;
+			}
+			else
+			{
+				num_of_frames = frm.additional_data.frame_number - frm.additional_data.last_frame_number;
+			}
+			auto diff = (double)(frm.additional_data.timestamp - frm.additional_data.last_timestamp) / (double)num_of_frames;
+			return diff > 0 ? std::max(1000.f / std::ceil(diff), (double)1) : frm.get_stream()->get_framerate();
+		}
+	};
+
+
+    class md_attribute_actual_fps : public md_attribute_parser_base
```

Comment:
This seems like it needs to be ds5_md_... since it is rather specific to the D400